sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpInterceptor.kt

```diff
@@ -0,0 +1,61 @@
+package io.sentry.android.okhttp
+
+import io.sentry.Breadcrumb
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.SpanStatus
+import java.io.IOException
+import okhttp3.Interceptor
+import okhttp3.Response
+
+class SentryOkHttpInterceptor(
+    private val hub: IHub = HubAdapter.getInstance()
+) : Interceptor {
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        var request = chain.request()
+
+        val url = request.url.toString()
+        val method = request.method
+
+        // read transaction from the bound scope
+        val span = hub.span?.startChild("http.client", "$method $url")
+
+        val response: Response
+
+        var code = -1
+        try {
+            span?.toSentryTrace()?.let {
+                request = request.newBuilder().addHeader(it.name, it.value).build()
+            }
+            response = chain.proceed(request)
+            code = response.code
+        } catch (e: IOException) {
+            span?.throwable = e
+            throw e
+        } finally {
+            span?.finish(SpanStatus.fromHttpStatusCode(code, SpanStatus.INTERNAL_ERROR))
+        }
+
+        val breadcrumb = Breadcrumb.http(request.url.toString(), request.method, code)
+        request.body?.contentLength().ifHasValidLength {
+            breadcrumb.setData("requestBodySize", it)
+        }
+        response.body?.contentLength().ifHasValidLength {
+            breadcrumb.setData("responseBodySize", it)
+        }
+        hub.addBreadcrumb(breadcrumb)
+
+        // TODO: collect ideas
+        // https://github.com/square/okhttp/blob/master/okhttp-logging-interceptor/src/main/kotlin/okhttp3/logging/HttpLoggingInterceptor.kt
+        // TODO: add package to options? how do we get options? does it even make sense?
+        // sdkVersion?.addPackage("maven:io.sentry:sentry-android-timber", BuildConfig.VERSION_NAME)
```

Comment:
we can do it through `Hub#getOptions` but I somehow doubt it makes sense to add it for each and every instrumentation we do. I am not the one with access to reporting tools though.

Reply:
for debugging (when there are events, we know each packages were being used).
or usage of our packages to take decisions like dropping a version or stuff like this.

true, forgot about `Hub#getOptions`, let's add then :)

Reply:
Concerns:
- we would need to add it to all events and transactions
- since `packages` on `sdkVersions` is a list (not set) creating multiple interceptors would add multiple same packages

Reply:
yeah, lets keep without it for now then and only do it when the OkHttp integration is able to captureEvent then, which would be then an integration for now it just works as an interceptor.